### DISCUSSION TITLE
A few questions about board design

### DISCUSSION THREAD

**Discussion by neeekw** [phil-barrett/grblHAL-teensy-4.x#68] [Q&A] [answered]
Hi Phil

I'm planning to attempt to design my own breakout board for the teensy 4.0 based off your design. Partly as a challenge, and partly as I see you have had issues shipping to South Africa. I am using the teensy 4.0 as that was available, while the 4.1 is almost impossible to get currently.

As low cost options for PCB fabbing are almost non-existent in South Africa, I plan to fab the PCB myself, (either by milling or UV dryfilm+etching) and therefore I would like to try make my design a single layer. 

My first question, is the TLP283-4 a suitable alternative to the LVT846S ?

And second is on the firmware side, regarding shuffling of pins. I assume for instance the spindle PWM pin cannot be pins that do not support PWM, but are there any less obvious limitations as to which functions (step, dir, limit i/o etc) can be on which pins ?

Regards
Nic

**Comment by phil-barrett** [comment, marked as answer]
Yes, the TLP283 should work fine.

The spindle pin should stay the same (Pin 13).  There is a comment in the grblHAL source about this, iirc.  Otherwise, the iMXRT 1062 is very flexible and you can move things around a lot.  You will need to be careful of i2c ports if you are going to use them.

Also, there is a mapping file for a t4.0 based board in the grblHAL sources - src/T40X101_map.h.  This has known good pin assignments and is running on at least one machine.  I recommend you follow that unless there are issues.  I have uploaded a [layout I did for the Teensy 4.0 in eagle](https://github.com/phil-barrett/grblHAL-teensy-4.x/blob/master/teensy%204%20x171.zip).  I believe that T40X101_map.h is compatible with it.

**Reply by neeekw** [reply]
I had not planned on incorporating i2c, but may rethink that after seeing Terjeio beautiful pendant.. 
But I will read through the source files properly and look at that mapping file before attempting a proper design.

I have also looked at the layout you did for the Teensy 4.0 and used that as a starting point for some playing around I did 
for a possible single layer board design, but up to now, the most complicated board I have designed have been small 8-bit AVR
projects, and being self taught, I'm not sure if it is any good
![playing](https://user-images.githubusercontent.com/115580840/195220103-117d80d7-ac3d-49f1-8d18-ca2ee8117fc9.png)

**Reply by phil-barrett** [reply]
I'm sure you will do fine. None of the circuits are very demanding, all the hard stuff is done on the teensy. Just use fairly large traces (12 mil/0.3mm) and keep good separation. Don't try make a really tight board.  I used to make boards at home but gave it up with the advent of cheap pcbs from China.
